src/Aquifer.API/Endpoints/Resources/Content/AssignEditor/Endpoint.cs

```diff
@@ -142,22 +142,21 @@ private static void SetDraftVersionStatus(bool isTakingBackFromReviewPending,
         UserEntity userToAssign,
         ResourceContentVersionEntity draftVersion)
     {
-        if (userToAssign.Role is not UserRole.Manager || !Constants.ManagerReviewStatuses.Contains(originalStatus))
+        if (userToAssign.Role is not UserRole.Manager && !Constants.CompanyReviewStatuses.Contains(originalStatus))
         {
             var newStatus = isTakingBackFromReviewPending
```

Comment:
Ah here's the part I was thinking of. I think this logic may still be tweaked to get things working. My brain is a bit fuzzy on all the specifics but I think maybe the else if below should be above this conditional, then it should be:
- if isTakingBackFromReviewPending assign to CompanyReview
- else if status is ai draft complete or new, change to EditorReview
- else do nothing

Does that seem right?

Reply:
Maybe also need a conditional in there for when a publisher is sending back to a manager?

tl;dr: the user assignment always happens, but the status only changes when being sent back from publisher -> manager, being taken from pending review to manager, or first being assigned to a user.

Reply:
I think so. I'll refactor it

Reply:
Github refresh fast enough didn't see your last comment. I'll make sure this works as expected